### PR TITLE
Add pydantic

### PR DESCRIPTION
**Pull request by yaroslavMain**
I added the pydantic scheme.

### CODE REVIEW

**Review by rbren — APPROVED** [review]

**Comment by rbren** [comment]
@yaroslavMain looks like the tests are still unhappy. You can run `poetry run pytest ./tests` locally to test them on your machine

File: opendevin/schema/action.py [hunk 1]
```diff
@@ -1,54 +1,54 @@
-from enum import Enum
+from pydantic import BaseModel, Field
 
 
-class ActionType(str, Enum):
-    INIT = "initialize"
-    """Initializes the agent. Only sent by client.
+class ActionType(BaseModel):
+    INIT: str = Field(default="initialize")
+    """Initialiszes the agent. Only sent by client.
```
Comment (enyst) [inline opendevin/schema/action.py:6]:
```suggestion
    """Initializes the agent. Only sent by client.
```

File: opendevin/schema/observation.py [hunk 2]
```diff
@@ -1,31 +1,38 @@
...
     """The result of a search
     """
 
-    CHAT = 'chat'
+    CHAT: str = Field(default='chat')
     """A message from the user
     """
 
-    MESSAGE = 'message'
+    MESSAGE: str = Field(default='message')
```
Comment (enyst) [inline opendevin/schema/observation.py:31]: Is this necessary?

Comment (yaroslavMain) [reply opendevin/schema/observation.py:31]: Yes

Comment (enyst) [reply opendevin/schema/observation.py:31]: You're right, I misread.

File: opendevin/schema/action.py [hunk 3]
```diff
@@ -1,68 +1,74 @@
...
+    SUMMARIZE: str = Field(default='summarize')
 
-    ADD_TASK = 'add_task'
+    ADD_TASK: str = Field(default='add_task')
 
-    MODIFY_TASK = 'modify_task'
+    MODIFY_TASK: str = Field(default='modify_task')
 
-    NULL = 'null'
+    NULL: str = Field(default='null')
```
Comment (rbren) [inline opendevin/schema/action.py:59]:
```suggestion
```

File: opendevin/schema/action.py [hunk 4]
```diff
@@ -1,68 +1,72 @@
...
 
-    STOP = 'stop'
+    STOP: str = Field(default='stop')
     """Stops the task. Must send a start action to restart a new task.
     """
 
-    CHANGE_TASK_STATE = 'change_task_state'
+    CHANGE_TASK_STATE: str = Field(default='change_task_state')
+
+ActionType = ActionTypeSchema()
```
Comment (rbren) [inline opendevin/schema/action.py:72]:
```suggestion

ActionType = ActionTypeSchema()
```

**Pull request merged**